src/holofuel/pages/TransactionHistory/TransactionHistory.js

```diff
@@ -76,19 +89,22 @@ export default function TransactionsHistory ({ history: { push } }) {
     !loadingPendingTransactions &&
     !loadingCompletedTransactions
 
+  let firstCompletedLabel, firstCompletedTransaction
+  if (partitionByDate(filteredCompletedTransactions)[0]) {
```

Comment:
it's probably NBD performance wise, but I think partitionByDate is relatively expensive, so I would store the result in a variable rather than calling it multiple times.